inject/android.js

```diff
@@ -16,10 +31,34 @@ function initCode () {
         platform: processedConfig.platform
     })
 
+    processedConfig.messageSecret = messageSecret
     init(processedConfig)
 
-    // Not supported:
-    // update(message)
+    // @ts-ignore
+    window.legacyContentScopeScriptsMessageCallback = (message) => {
+        update(message)
+    }
+
+    window.addEventListener('sendMessageProxy' + messageSecret, event => {
+        event.stopImmediatePropagation()
+
+        if (!(event instanceof CustomEvent) || !event?.detail) {
+            return console.warn('no details in sendMessage proxy', event)
+        }
+
+        const messageType = event.detail?.messageType
+        if (!allowedMessages.includes(messageType)) {
+            return console.warn('Ignoring invalid sendMessage messageType', messageType)
+        }
+
+        const message = {
+            type: messageType,
+            options: event.detail?.options
+        }
+        const stringifiedArgs = JSON.stringify(message)
+        // @ts-ignore
+        window.legacyContentScopeScriptsMessageHandler.process(stringifiedArgs)
```

Comment:
Yes, this is used for sending messages back to the app.
To do this I use the `addJavascriptInterface()` method on `WebView` to add the interface.

I don't have any control where it puts this unfortunately. This is from the WebView docs:
_"Because the object is exposed to all the frames, any frame could obtain the object name and call methods on it. There is no way to tell the calling frame's origin from the app side, so the app must not assume that the caller is trustworthy unless the app can guarantee that no third party content is ever loaded into the WebView even inside an iframe."_

So what I can do is check the secret on the other side and if it doesn't match then ignore the message.

Reply:
OK got it. If possible, I would:
1. Put this directly in the `window` Object, instead of `window.legacyContentScopeScriptsMessageHandler`.
2. Give it a random name, that's then also passed over from the Android code with the other secrets.

Passing the secret into the function makes sense, but we do need to make sure the website can't change the function. If something like `window.legacyContentScopeScriptsMessageHandler.process = evilFunction` is possible, the secret could get intercepted. `Object.getOwnPropertyDescriptor(window.legacyContentScopeScriptsMessageHandler, 'process')` should tell us if that's configurable or writable. Alternatively, something like `window.legacyContentScopeScriptsMessageHandler.process = alert` could be a quick test.

Reply:
1. isn't possible
2. is possible :)

> If something like window.legacyContentScopeScriptsMessageHandler.process = evilFunction is possible

iirc we never checked if this wrapping is possible; see "Spoofing communication boundaries" under https://app.asana.com/0/944732614786175/1202203984294184/f I think assignment was already checked but there's other ways listed there.

Reply:
1. Isn't possible as Jonathan stated, we can only add the interface onto the `window`
2. I've updated the code to pass in a random name from Android 👍

@kzar I ran `Object.getOwnPropertyDescriptor(window.legacyContentScopeScriptsMessageHandler, 'process')` and result is `{"writable":true,"enumerable":true,"configurable":true}`. Let me know if there's anything else I can do here.

Reply:
I wonder if we can/should take the windows approach of grabbing the handler then deleting it.